Initial commit

knitHacker opened the pull request:
Can install it locally and import the module successfully.

markrawls commented on tests/test_image_adding.py:
```diff
@@ -0,0 +1,50 @@
+import pytest
```
Not a change request, but +200 points to you for using `pytest` and including unit tests in v1

knitHacker replied:
hahaha don't give me credit for it. it was with the initial code in doc-gen

markrawls commented on signhere/utils.py:
```diff
@@ -0,0 +1,277 @@
+from .exceptions import AddImageArgumentError
+
+
+DEFAULT_SCALAR = 0.75
```
Dumb question - what's this for?

cviveiro replied:
i found that this was the scaling needed in the x and y direction to get things placed correctly. this is scaling from the (x, y) coords we get from pdf gen to the (x, y) coords we need to use to place the picture in the right place on the pdf

markrawls reviewed (changes requested):
Also I'm about 80% confident we can't (and 100% certain we shouldn't) legally host the Century Expanded Bold font file here. That and I don't think the general public using this library would be needing that. Overall though, coming along nicely!!

knitHacker commented on README.md:
````diff
@@ -0,0 +1,36 @@
+## Usage
+
+```python
+import signhere.utils
+signhere.utils.
````
haha sorry i didn't actually know how to use it so i stopped in the middle of the example.

knitHacker replied:
just to clarify you want to ditch the README.md and substitute something generated by sphinx? so I would check in the generate documentation?

markrawls replied:
Yep! I'd say check in the source files and set up a ReadTheDocs page for it, don't commit the generated documentation.

knitHacker commented:
Thanks! Will remove the font and make suggested code changes. So far I hadn't really touched the code other than update the import in tests.

knitHacker commented on signhere/utils.py:
```diff
@@ -0,0 +1,277 @@
+    dynamic_text,
+    only_matches=None,
+    page_numbers=None,
+    loader_args=None,
```
do you want type annotations on all function arguments or just the public ones?

markrawls replied:
Ideally all of them, but if it becomes too cumbersome then we can ditch annotating private members.

knitHacker commented:
@markrawls would appreciate another look through. I cleaned up the code / added in sphinx type docstrings and added in sphinx docs. Still have the readme because not sure how / if I should use generated (and checked in) sphinx docs for the readme

markrawls commented on docs/conf.py:
```diff
@@ -0,0 +1,51 @@
+# -- Project information -----------------------------------------------------
+
+project = "signhere"
+copyright = "2021, Mark Rawls"
+author = "Mark Rawls"
```
This should probably be your name (or the company's) instead of mine

knitHacker replied:
I can put the companies name but all the actual code was written by you and christine wrote all the actual code haha

markrawls reviewed (changes requested):
Outside of that, looks good!!

knitHacker commented:
Glad I went to add type annotations to the private functions because I realized I didn't remove the references to the IMAGE_CONSTANTS dict that I removed.

markrawls reviewed (approved):
This is beautiful. Great job on that! We should make a ReadTheDocs org (if that's a thing they offer) and get our docs for this hosted there, but this is more than enough to get started.
